app/lib/account/default_auth_provider.dart

```diff
@@ -35,6 +35,78 @@ class DefaultAuthProvider extends AuthProvider {
     _oauthApi = oauth2_v2.Oauth2Api(_httpClient);
   }
 
+  @override
+  Future<void> close() async {
+    _httpClient.close();
+  }
+
+  @override
+  Future<oauth2_v2.Tokeninfo> callTokenInfoWithAccessToken({
+    required String accessToken,
+  }) async {
+    return _oauthApi.tokeninfo(accessToken: accessToken);
+  }
+
+  @override
+  Future<http.Response> callTokenInfoWithIdToken(
+      {required String idToken}) async {
+    // Hit the token-info end-point documented at:
+    // https://developers.google.com/identity/sign-in/web/backend-auth
+    // Note: ideally, we would verify these JWTs locally, but unfortunately
+    //       we don't have a solid RSA implementation available in Dart.
+    final u =
+        _tokenInfoEndPoint.replace(queryParameters: {'id_token': idToken});
+    return await _httpClient.get(u, headers: {'accept': 'application/json'});
+  }
+
+  @override
+  Future<oauth2_v2.Userinfo> callGetUserinfo(
+      {required String accessToken}) async {
+    final authClient = auth.authenticatedClient(
+        _httpClient,
+        auth.AccessCredentials(
+          auth.AccessToken(
+            'Bearer',
+            accessToken,
+            clock.now().toUtc().add(Duration(minutes: 20)), // avoid refresh
+          ),
+          null,
+          [],
+        ));
+
+    final oauth2 = oauth2_v2.Oauth2Api(authClient);
+    return await oauth2.userinfo.get();
+  }
+
+  @override
+  Future<bool> verifyTokenSignature({
+    required JsonWebToken token,
+    required Future<OpenIdData> Function() openIdDataFetch,
+  }) async {
+    final openIdData = await openIdDataFetch();
+    return await token.verifySignature(openIdData.jwks);
```

Comment:
We need caching here, right?

Perhaps we should pass `OpenIdData` and do the actual caching outside `auth_provider`?

Reply:
The referenced/passed-in `openIdDataFetch()` functions do the caching.